packages/flutter/lib/src/gestures/events.dart

```diff
@@ -305,11 +344,44 @@ abstract class PointerEvent extends Diagnosticable {
   /// the difference between the 2 events in that case.
   final bool synthesized;
 
+  /// The transformation used to transform this event from the global coordinate
+  /// space into the coordinate space of the event receiver.
+  ///
+  /// This value affects what is returned by [localPosition] and [localDelta].
+  /// If this value is null, it is treated as the identity transformation.
+  ///
+  /// Unlike a paint transform, this transform usually does not contain any
+  /// "perspective" components, meaning that the third row and the third column
+  /// of the matrix should be equal to "0, 0, 1, 0". This ensures that
+  /// [localPosition] describes the point in the local coordinate system of the
+  /// event receiver at which the user is actually touching the screen.
+  ///
+  /// See also:
+  ///
+  ///  * [transformed], which transforms this event into a different coordinate
+  ///    space.
+  final Matrix4 transform;
+
+  /// The original [PointerEvent] before it was transformed by [transform].
+  final PointerEvent original;
+
+  /// Transforms the event from the global coordinate space into the coordinate
+  /// space of an event receiver.
+  ///
+  /// The coordinate space of the event receiver is described by `transform`. A
+  /// null value for `transform` is treated as the identity transformation.
+  ///
+  /// The method may return the same object instance if for example the
+  /// transformation has no effect on the event.
+  PointerEvent transformed(Matrix4 transform);
```

Comment:
These docs should be clearer that this is not cumulative, that this replaces [transform].